src/main/scala/cromwell/binding/values/WdlFileLike.scala

```diff
@@ -0,0 +1,10 @@
+package cromwell.binding.values
+
+import cromwell.binding.types.WdlFileType
+
+/**
+ * WDL value which identifies the location of a File-like object.
+ */
+trait WdlFileLike extends WdlPrimitive {
+  val wdlType = WdlFileType
```

Comment:
Adding a note here just because I'm likely to forget when we discuss this. We need a way so that in WDL a user can specify a task input as being `File` but it might be a GCS path. As the code currently exists it briefly appears to work but what happens is that in `WdlFile.apply()` that GCS path loses one of its slashes as it passes through `Paths.get` and then things die 10-15 minutes later when JES decides to let us know a job completes :(

I'm still fuzzy on the how some of these pieces fit together though so want that to be part of this discussion as well as it seems related.